assets/bulk-add.js

```diff
@@ -0,0 +1,37 @@
+class BulkAdd extends HTMLElement {
+  constructor() {
+    super();
+    this.queue = []
+    this.requestStarted = false;
+  }
+
+  startQueue(id, quantity) {
+    this.queue.push({id, quantity})
+    const int = setInterval(() => {
+      if (this.queue.length > 0) {
+        if (!this.requestStarted)  {
+          this.sendRequest(this.queue)
+        }
+      } else {
+        clearInterval(int)
+      }
+    }, 250)
+  }
+
+
+  sendRequest(queue) {
+    this.requestStarted = true;
+    const items = {}
+    const ids = []
+    queue.forEach((q) => {
+      items[parseInt(q.id)] = q.quantity;
+    });
+    queue.forEach((q) => {
+      ids.push(q.id)
+    })
```

Comment:
These don't need to be separate `forEach` loops, you can do the `items` assignment and the `ids` push in the same iteration.

Reply:
good idea :)